### PR TITLE
Disable document outline

### PR DESCRIPTION
**Pull request by jmarolf**
workaround while we improve performance in https://github.com/dotnet/roslyn/pull/65641

CC: @allisonchou this a fix we want to get into the next servicing release.

### CODE REVIEW

File: src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs:239]: We don't get called if the user never had the document outline window open in the first place, right? This is just a mitigation for those who had it open for other languages and didn't notice?

File: src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs:278]: What happens if the user changes the option after we created one? Wouldn't this then leak?

File: src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs
Comment (jmarolf) [reply src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs:239]: That is correct

File: src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs
Comment (jmarolf) [reply src/VisualStudio/Core/Def/LanguageService/AbstractLanguageService`2.VsCodeWindowManager.cs:278]: Good point, we should just always dispose these if they are not null

**Comment by CyrusNajmabadi** [comment]
This feels weird to me.  Say someone was using the feature.  So they have the outlining window open.  Now VS will launch and this will just be empty.  Won't they think something is broken here?  How would they know they need to enable this feature to get things working again?

**Comment by jmarolf** [comment]
> This feels weird to me. Say someone was using the feature. So they have the outlining window open. Now VS will launch and this will just be empty. Won't they think something is broken here? How would they know they need to enable this feature to get things working again?

https://github.com/dotnet/roslyn/pull/65641 is expected to be the final resolution to this. this is a stop gap to unblock users wbo have the document outline window opened for other reasons (such as XAML) and who currently have no other recourse.

**Comment by CyrusNajmabadi** [comment]
I guess my point is: can we make this less confusing for users who run into this?  For example, we could put a message (or button) in the ui explaining that this is not working for C#/VB until the option is explicitly enabled.

**Comment by jmarolf** [comment]
> I guess my point is: can we make this less confusing for users who run into this? For example, we could put a message (or button) in the ui explaining that this is not working for C#/VB until the option is explicitly enabled.

Not a bad idea, I'll ask what the rules are for getting a new translation in for that text. Question would be what we do if the translation team couldn't turn it around in time for the servicing release. Would we want to wait for the next servicing release? I don't think we are allowed to release un-translated strings in a servicing patch.

**Comment by jmarolf** [comment]
> I guess my point is: can we make this less confusing for users who run into this? For example, we could put a message (or button) in the ui explaining that this is not working for C#/VB until the option is explicitly enabled.

At this point we can't add new strings in a servicing update. I am open to options here but I think _if_ we want to insert this now we should do so as is.

**Comment by jmarolf** [comment]
@arkalyanms I believe integration tests are broken in the release/dev17.4 branch so I assume this is good to go once its reviewed

**Comment by jmarolf** [comment]
@jasonmalinowski We've decided we'd like to take this change for servicing. Can you review?

**Review by arkalyanms — APPROVED** [review]

File: src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineOptionsMetadata.cs
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineOptionsMetadata.cs:14]: so we're disabling the feature entirely?  there's no UI to get it back?

**Review by jasonmalinowski — APPROVED** [review]
😢

File: src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineOptionsMetadata.cs
Comment (jmarolf) [reply src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineOptionsMetadata.cs:14]: that is correct

**Pull request merged**
